src/configuration/configuration.ts

```diff
@@ -170,6 +174,17 @@ class Configuration implements IConfiguration {
     );
   }
 
+  unproxify(obj: Object): Object {
```

Comment:
I don't quite understand why this is necessary.

Reply:
I removed that piece of code and made another change: https://github.com/VSCodeVim/Vim/commit/10409227b091c5f72c7326e6894127f014c37d70. All tests passed?

Reply:
With the changes here https://github.com/VSCodeVim/Vim/commit/10409227b091c5f72c7326e6894127f014c37d70, I was able to get your tests to pass.

Reply:
I don't quite follow what's going on. 

> The vscode.workspace.getConfiguration() function return a Proxy object instead of a object. The Proxy object's set method is replaced.

We never should be calling the Proxy object's set method.

```
    let vimConfigs: any = Globals.isTesting
      ? Globals.mockConfiguration
      : this.getConfiguration('vim');

    /* tslint:disable:forin */
    // Disable forin rule here as we make accessors enumerable.`
    for (const option in this) {
      const val = vimConfigs[option] as any;
      if (val) {
        this[option] = val;
      }
    }
```

`this[option] = val` is calling the proxy object's getter here. 

Additionally, we already have a pattern for setting configurations in this nested manner (see `vim.debug.logginglevel`) and I just submitted a PR to move some other configurations to the same pattern (https://github.com/VSCodeVim/Vim/pull/2915). Haven't come across the same error you are seeing.

Reply:
I have search the whole project, the `vim.debug.logginglevel` only appear in package.json and README.md. I haven't found the logic like `vim.debug.logginglevel = xxx1` in code.
I added `vim.debug.logginglevel = 'warn'` in project and I got the same error:
```
TypeError: Cannot assign to read only property 'loggingLevel' of object。
Activating extension `vscodevim.vim` failed:  TypeError: Cannot assign to read only property 'loggingLevel' of object
```
You can try it yourself to see if you will get the same error!

Reply:
Ok, I see the issue now. Thanks for explaining.